src/common/reducers/repository-input.js

```diff
@@ -19,39 +13,15 @@ export function repositoryInput(state = {
       return {
         ...state,
         inputValue: action.payload,
-        repository: parseRepository(action.payload),
+        repository: parseGitHubRepoUrl(action.payload),
```

Comment:
I think this is fine. On the last React app we worked on, forms were a common part of our stores, and recurring properties like "account number", "amount", "credit card number" and "password" appeared in quite a few places. We extracted common code (usually functions for validating, transforming and decorating data) into helper functions as much as possible. Standardising the shape of the data made this easier.